src/controllers/EvidenceUploadController.ts

```diff
@@ -52,11 +52,10 @@ const continueButtonValidator: Validator = {
 
     validate(request: Request): ValidationResult {
 
-        const appeal: Appeal = request.session
-            .chain(_ => _.getExtraData())
-            .map<ApplicationData>(data => data[APPLICATION_DATA_KEY])
-            .map(data => data.appeal)
-            .unsafeCoerce();
+        const applicationData: ApplicationData = request.session!
+            .getExtraData(APPLICATION_DATA_KEY) || {} as ApplicationData;
+
+        const appeal: Appeal = applicationData!.appeal;
```

Comment:
throw an exception if the appeal is missing. And I don't think you need the ! since the value of applicationData is always defined